### ISSUE TITLE
Strange RGB colors after applying NBAR

### ISSUE THREAD

**Issue by nbennasr** [MarcYin/SIAC#27]
Can you please take a look at the result of NBAR applied on the image below where RGB colors seem strange:
id: 9c95c2cf-ce9a-4dd1-b29c-c065f10a119a
name: downloads/S2B_MSIL1C_20220821T233649_N0400_R030_T60WWT_20220821T235303
mosaic periode: [2022-08-01  2022-08-31]

SIAC
![image](https://github.com/MarcYin/SIAC/assets/116064092/a298264a-48df-4b7e-add7-d833864587d1)

SIAC+NBAR
![image](https://github.com/MarcYin/SIAC/assets/116064092/632e20b7-25df-4a9a-ac20-f5718b71a253)

**Comment by MarcYin** [comment]
Hi,

I tried with the same image and it seems to be ok for me?

SIAC:


![Screenshot 2023-08-17 at 17 28 12](https://github.com/MarcYin/SIAC/assets/19539641/e0b09fe7-bcf6-4e0c-a3d2-0d58264be54f)

SIAC+ NBAR

![Screenshot 2023-08-17 at 17 28 26](https://github.com/MarcYin/SIAC/assets/19539641/91ce68d0-f0b8-4d08-92db-b2e6cdb6d101)


I run it with:

```python
get_nbar.create_nbar('S2B_MSIL1C_20220821T233649_N0400_R030_T60WWT_20220821T235303.SAFE', mosaic_start_date = datetime(2022, 8, 1), mosaic_end_date=datetime(2022, 8, 31), Gee=False, use_VIIRS=True, nbar_sza = 'temporal_average_sza')
```

**Comment by nbennasr** [comment]
Hi @MarcYin

I run with the same parameters as you mentioned and I get the result of NBAR with strange colors.
The last commit that I clone in my repo is 8afc2c3524e19df27cd7e906a7ce03b822896efd

SIAC:
![image](https://github.com/MarcYin/SIAC/assets/116064092/bfab7638-89be-45a2-a1cc-44fdbf598cd7)

SIAC + NBAR:
![image](https://github.com/MarcYin/SIAC/assets/116064092/fadd6aa7-f307-4a6e-acd6-87d1fe180612)

**Comment by MarcYin** [comment]
Yes, I have the same version. Could you please share the file so that I can check the intermediate outputs?

**Comment by nbennasr** [comment]
Thank you @MarcYin for your quick replay!
You can find my output [here](https://ftp.luxcarta.online/S2B_MSIL1C_20220821T233649_N0400_R030_T01WCN_20220821T235303.SAFE.zip)

RGB band of SIAC + NBAR are:
T01WCN_20220821T233649_B04_nbar_sza_temporal_avg.tif  
T01WCN_20220821T233649_B03_nbar_sza_temporal_avg.tif  
T01WCN_20220821T233649_B02_nbar_sza_temporal_avg.tif

**Comment by MarcYin** [comment]
Hi,

I can confirm this issue and it has been fixed in [6a41b20](https://github.com/MarcYin/SIAC/commit/6a41b2040ffb2b4f832cf5ffeda4f2609d28b426). It is caused by the calculation of mosaic hour crossing the antimeridian.

**Comment by nbennasr** [comment]
Thank you @MarcYin !
